plugins/language_st.lua

```diff
@@ -0,0 +1,127 @@
+-- mod-version:3
+local syntax = require "core.syntax"
+
+-- Language syntax references
+-- https://pdhonline.com/courses/e334/e334content.pdf
+
+syntax.add {
+  name = "PLC Structured Text IEC 61131-3",
+  files = { "%.stx?$", "%.iecst$" },
+  comment = "//",
+  block_comment = { "%(%*", "%*%)" },
+  patterns = {
+    { pattern = "//.*",                                     type = "comment"  }, -- Single-line comment
+    { pattern = { "%(%*", "%*%)" },                         type = "comment"  }, -- Multi-line comment
+    { pattern = { '"', '"', '\\' },                         type = "string"   }, -- String, quotation marks
+    { pattern = { "'", "'", '\\' },                         type = "string"   }, -- String, apices
+    { regex   = "\\s\\w+\\#[0-9]+[m|s]?",                   type = "number"   }, -- Time/Date formats
+    { regex   = "(?>[^(\\W]+)(?=\\()",                      type = "function" }, -- Function
+    { pattern = "[%a_][%w_]*",                              type = "symbol"   }, -- Symbols
+    { regex   = "^\\s*[A-Z]+_[A-Z]*\\s?\\;?\\s?",           type = "keyword"  }, -- Keyword
+    { regex   = "\\:\\s*\\w+",                              type = "keyword2" }, -- Variable/Method Type
+    { regex   = "(?:\\-|\\+)?\\d+\\.*(?:\\-|\\+)?\\d+",     type = "number"   }, -- Number Range
+    { pattern = "[%+%-=/%*%^%%<>!~|&:]",                    type = "operator" }, -- Operators
+    { pattern = "-?0x%x+",                                  type = "number"   }, -- Number
+    { pattern = "-?%d+[%deE]*f?",                           type = "number"   }, -- Number
+    { pattern = "-?%.?%d+f?",                               type = "number"   }, -- Number
+    { regex   = "^\\w+()\\s(?:[A-Z]+[a-z]*\\_?)+(?=\\w*)",  type = { "normal", "function" } }, -- Function implementation
```

Comment:
```suggestion
    { regex   = [[^\w+()\s(?:[A-Z]+[a-z]*\_?)+(?=\w*)]],    type = { "normal", "function" } }, -- Function implementation
```
Should this allow for spaces at the start?
What about allowing multiple spaces between the first `\w+` and the group?
What is the group trying to match?
What is the lookahead at the end doing?

Reply:
This pattern as well, for some reason removing it doesn't affect any highlighted example of code.